src/Symfony/Component/Console/Application.php

```diff
@@ -855,7 +855,7 @@ protected function configureIO(InputInterface $input, OutputInterface $output)
             $input->setInteractive(false);
         } elseif (function_exists('posix_isatty') && $this->getHelperSet()->has('dialog')) {
             $inputStream = $this->getHelperSet()->get('dialog')->getInputStream();
-            if (!@posix_isatty($inputStream)) {
+            if (!@posix_isatty($inputStream) && true !== getenv('SHELL_INTERACTIVE')) {
```

Comment:
``getenv`` can return a string or ``false`` when the env var is not set. But it cannot be ``true``